applicationset/controllers/applicationset_controller.go

```diff
@@ -1433,17 +1433,37 @@ func (r *ApplicationSetReconciler) setAppSetApplicationStatus(ctx context.Contex
 	needToUpdateStatus := false
 
 	if len(applicationStatuses) != len(applicationSet.Status.ApplicationStatus) {
+		logCtx.Debugf("app status count %d→%d",
```

Comment:
should we do something like for all the log lines? (feel free to use better keys that are more meaningful)
```
logCtx.WithFields(log.Fields{
	"status_count":   len(applicationSet.Status.ApplicationStatus),
	"expected_count": len(applicationStatuses),
}).Debug("application status count")

if i was looking at these debug logs i wouldnt know what the numbers represent.